service/s3/s3manager/batch_test.go

```diff
@@ -590,8 +592,141 @@ func TestBatchError(t *testing.T) {
 		t.Error("Expected error, but received nil")
 	}
 
-	if svc.index != len(objects) {
-		t.Errorf("Expected %d, but received %d", len(objects), svc.index)
+	if index != len(objects) {
+		t.Errorf("Expected %d, but received %d", len(objects), index)
+	}
+
+}
+
+type testAfterIter struct {
+	afterDelete   bool
+	afterDownload bool
+	afterUpload   bool
+	index         int
+}
+
+func (iter *testAfterIter) Next() bool {
+	next := (iter.index & 1) == 0
```

Comment:
Nit this will only wouldn't this be similar to just do something like the following, or change from an index, to a boolean flag.

```go
func (iter *testAfterIter) Next() bool {
    defer func() { iter.index++ }
    return iter.index == 0
}
```